### PR TITLE
wai-extra: Fix Network.Wai.Test.request always sending an empty request body 

### PR DESCRIPTION
**Pull request by mheinzel**
Before submitting your PR, check that you've:

- [x] Bumped the version number

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

When trying to find out why request bodies went missing in our test suite, I noticed that `request` just calls `srequest` with an empty request body. This drops the existing body, which is not the behaviour I expected.

The first commit updates the tests to catch the problem:

```
  srequest
    returns the response body    of an echo app
  request
    returns the status code      of an echo app on default request
    returns the response body    of an echo app FAILED [1]
    returns the response headers of an echo app

Failures:

  test/Network/Wai/TestSpec.hs:98:7:
  1) Network.Wai.Test.request returns the response body    of an echo app
       expected: "request body"
        but got: ""

  To rerun use: --match "/Network.Wai.Test/request/returns the response body    of an echo app/"
```

I'm not sure if anyone is relying on the current behaviour, so please let me know if just increasing the patch number is no enough.

### CODE REVIEW

File: wai-extra/ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog for wai-extra
 
+## 3.0.29.1
+
+* Fix `Network.Wai.request` always sending an empty request body [#794](https://github.com/yesodweb/wai/pull/794)
```
[the remark this answers is not in the document]
Comment (mheinzel) [reply wai-extra/ChangeLog.md]: You are right, thanks! Fixed.

**Comment by snoyberg** [comment]
Thanks!
